Microsoft.Toolkit.Uwp.UI/Behaviors/ViewportBehavior.cs

```diff
@@ -0,0 +1,205 @@
+        /// </summary>
+        public event EventHandler EnteringViewport;
+
+        /// <summary>
+        /// Associated element exit the ScrollViewer viewport event
+        /// </summary>
+        public event EventHandler ExitingViewport;
+
+        /// <summary>
+        /// Gets or sets a value indicating whether remove this behavior after associated element entered viewport.
+        /// </summary>
+        public bool IsAlwaysOn
+        {
+            get { return (bool)GetValue(IsAlwaysOnProperty); }
+            set { SetValue(IsAlwaysOnProperty, value); }
+        }
+
+        /// <summary>
+        /// Gets a value indicating whether associated element is fully in the ScrollViewer viewport
+        /// </summary>
+        public bool IsFullyInViewport
+        {
+            get { return (bool)GetValue(IsFullyInViewportProperty); }
+            private set { SetValue(IsFullyInViewportProperty, value); }
+        }
+
+        /// <summary>
+        /// Gets a value indicating whether associated element is in the ScrollViewer viewport
+        /// </summary>
+        public bool IsInViewport
+        {
+            get { return (bool)GetValue(IsInViewportProperty); }
+            private set { SetValue(IsInViewportProperty, value); }
+        }
+
+        /// <summary>
+        /// Called after the behavior is attached to the <see cref="P:Microsoft.Xaml.Interactivity.Behavior.AssociatedObject" />.
+        /// </summary>
+        /// <remarks>
+        /// Override this to hook up functionality to the <see cref="P:Microsoft.Xaml.Interactivity.Behavior.AssociatedObject" />
+        /// </remarks>
+        protected override void OnAttached()
+        {
+            base.OnAttached();
+
+            var parent = VisualTreeHelper.GetParent(AssociatedObject);
+            if (parent == null)
+            {
+                AssociatedObject.Loaded += AssociatedObject_Loaded;
+                return;
+            }
+
+            Init();
+        }
+
+        /// <summary>
+        /// Called when the behavior is being detached from its <see cref="P:Microsoft.Xaml.Interactivity.Behavior.AssociatedObject" />.
+        /// </summary>
+        /// <remarks>
+        /// Override this to unhook functionality from the <see cref="P:Microsoft.Xaml.Interactivity.Behavior.AssociatedObject" />
+        /// </remarks>
+        protected override void OnDetaching()
+        {
+            base.OnDetaching();
+
+            _hostScrollViewer.ViewChanged -= ParentScrollViewer_ViewChanged;
+            _hostScrollViewer = null;
+        }
+
+        private static void OnIsFullyInViewportChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            var obj = (ViewportBehavior)d;
+            var value = (bool)e.NewValue;
+            if (value)
+            {
+                obj.EnteredViewport?.Invoke(obj.AssociatedObject, EventArgs.Empty);
+
+                if (obj.IsAlwaysOn)
```

Comment:
Since we changed the name to mean the opposite, we need to invert the logic using it here too, right?

```suggestion
                if (!obj.IsAlwaysOn)
```

Reply:
Done.